File: Architecture.Core/Entities/NotificationInfo.cs
using Architecture.Core.Entities.Core;
using System;
using System.Collections.Generic;
using System.Text;

namespace Architecture.Core.Entities.Notification
{
    public class NotificationInfo: Auditable
    {
        public int NotificationInfoId { get; set; }
        public string MessageContent { get; set; }
        public int? ParentMessageId { get; set; }
        public bool IsGlobal { get; set; }
        public Guid? MessageFor { get; set; }
        public bool IsSeen { get; set; }
        public DateTime? SeenTime { get; set; }
        public int? OfferInfoId { get; set; }
        public virtual OfferInfo OfferInfo { get; set; }
        public string Type { get; set; }
    }
}

File: Architecture.Web/ClientApp/src/app/Shared/Entity/Notifiation/NotificationInfo.ts
import { Auditable } from '../Common/auditable';
import { OfferInfo } from '../Users/OfferInfo';

export class NotificationInfo extends Auditable {
    public notificationInfoId: number;
    public messageContent: string;
    public parentMessageId: string;
    public isGlobal: boolean = false;
    public messageFor: string;
    public isSeen: boolean=false;
    public seenTime: Date;
    public offerInfoId: number;
    public type: string;
    public offerInfo: OfferInfo;
    public createdByName: string;
    public messageForName: string;
}

